source/heat/laser_heat_source_gauss.cpp

```diff
@@ -20,15 +22,118 @@ namespace MeltPoolDG::Heat
+
+    return data.absorptivity * projection_factor * delta_value *
+           power_density_interface(distance, power);
+  }
+
+  template <int dim>
+  void
+  LaserHeatSourceGauss<dim>::compute_interface_heat_source(VectorType &heat_source_vector,
+                                                           const ScratchData<dim> &scratch_data,
+                                                           const unsigned int      temp_dof_idx,
+                                                           const double            laser_power,
+                                                           const Point<dim> &      laser_position,
+                                                           const VectorType & level_set_heaviside,
+                                                           const unsigned int ls_dof_idx,
+                                                           const bool         zero_out) const
+  {
+    if (zero_out)
+      scratch_data.initialize_dof_vector(heat_source_vector, temp_dof_idx);
+
+    FEValues<dim> heat_source_eval(
+      scratch_data.get_mapping(),
+      scratch_data.get_dof_handler(temp_dof_idx).get_fe(),
+      Quadrature<dim>(
+        scratch_data.get_dof_handler(temp_dof_idx).get_fe().get_unit_support_points()),
+      update_quadrature_points);
+
+    level_set_heaviside.update_ghost_values();
+    FEValues<dim> ls_heaviside_eval(
+      scratch_data.get_mapping(),
+      scratch_data.get_dof_handler(ls_dof_idx).get_fe(),
+      Quadrature<dim>(scratch_data.get_dof_handler(ls_dof_idx).get_fe().get_unit_support_points()),
+      update_gradients);
+
+    const unsigned int dofs_per_cell =
+      scratch_data.get_dof_handler(temp_dof_idx).get_fe().n_dofs_per_cell();
+    std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+    // TODO: find a better way to interpolate the (level-set dependent) laser heat source onto the
+    // heat source dof vector. Problem: if ls-degree=1, its gradient may be discontinuous!
+    for (const auto &cell : scratch_data.get_dof_handler(temp_dof_idx).active_cell_iterators())
+      {
+        if (cell->is_locally_owned())
+          {
+            cell->get_dof_indices(local_dof_indices);
+
+            heat_source_eval.reinit(cell);
+            ls_heaviside_eval.reinit(cell);
+
+            std::vector<dealii::Tensor<1, dim, double>> grad_ls_heaviside(
+              ls_heaviside_eval.n_quadrature_points);
+            ls_heaviside_eval.get_function_gradients(level_set_heaviside, grad_ls_heaviside);
+
+            for (const auto q : heat_source_eval.quadrature_point_indices())
+              {
+                const double delta_value = grad_ls_heaviside[q].norm();
+                if (delta_value == 0.0)
+                  {
+                    heat_source_vector[local_dof_indices[q]] = 0.0;
+                    continue;
+                  }
+                const Tensor<1, dim, double> normal_vector = grad_ls_heaviside[q] / delta_value;
```

Comment:
Here, we might face problems since we are not using the diffused `normal_vector`.

```suggestion
// TODO: use computed normal vector
                const Tensor<1, dim, double> normal_vector = grad_ls_heaviside[q] / delta_value;
```